tests/migration_test.go

```diff
@@ -62,39 +60,41 @@ const (
 
 var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system] VM Live Migration", func() {
 	var virtClient kubecli.KubevirtClient
-
-	var originalKubeVirtConfig *k8sv1.ConfigMap
 	var err error
 
-	tests.BeforeAll(func() {
+	var originalKubeVirtConfig *v1.KubeVirt
 
+	tests.BeforeAll(func() {
 		virtClient, err = kubecli.GetKubevirtClient()
 		tests.PanicOnError(err)
 
-		originalKubeVirtConfig, err = virtClient.CoreV1().ConfigMaps(tests.KubeVirtInstallNamespace).Get(virtconfig.ConfigMapName, metav1.GetOptions{})
-		if err != nil && !errors.IsNotFound(err) {
-			Expect(err).ToNot(HaveOccurred())
-		}
+		originalKubeVirtConfig = tests.GetCurrentKv(virtClient)
+	})
 
-		if errors.IsNotFound(err) {
-			// create an empty kubevirt-config configmap if none exists.
-			cfgMap := &k8sv1.ConfigMap{
-				ObjectMeta: metav1.ObjectMeta{Name: virtconfig.ConfigMapName},
-				Data: map[string]string{
-					"feature-gates": "",
-				},
-			}
+	defaultKVConfig := func() v1.KubeVirtConfiguration {
+		kvc := originalKubeVirtConfig.DeepCopy()
 
-			originalKubeVirtConfig, err = virtClient.CoreV1().ConfigMaps(tests.KubeVirtInstallNamespace).Create(cfgMap)
-			if err != nil {
-				Expect(err).ToNot(HaveOccurred())
-			}
+		if kvc.Spec.Configuration.MigrationConfiguration == nil {
+			kvc.Spec.Configuration.MigrationConfiguration = &v1.MigrationConfiguration{}
+		}
 
+		if kvc.Spec.Configuration.DeveloperConfiguration == nil {
+			kvc.Spec.Configuration.DeveloperConfiguration = &v1.DeveloperConfiguration{}
 		}
-	})
+
+		if kvc.Spec.Configuration.NetworkConfiguration == nil {
+			kvc.Spec.Configuration.NetworkConfiguration = &v1.NetworkConfiguration{}
+		}
+
+		return kvc.Spec.Configuration
+	}
 
 	BeforeEach(func() {
 		tests.BeforeTestCleanup()
+
+		err = virtClient.CoreV1().ConfigMaps(tests.KubeVirtInstallNamespace).Delete(virtconfig.ConfigMapName, &metav1.DeleteOptions{})
```

Comment:
don't we need to restore this configMap after the test? Otherwise we're permanently mutating the kubevirt install, right?